ha/cli/exec/clusterExecutor.py

```diff
@@ -129,50 +132,105 @@ def execute(self):
 
-    def execute(self) -> str:
+    def execute(self) -> None:
         raise HAUnimplemented("This operation is not implemented.")
 
-class ClusterAddExecutor(CommandExecutor):
-    def validate(self) -> str:
-        raise HAUnimplemented("This operation is not implemented.")
+class ClusterNodeAddExecutor(CommandExecutor):
+    '''
+        Module which will accept the request for cluster add node
+        functinality and which is responsible for delegating that request
+        to Cluster Manager
+    '''
 
-    def execute(self) -> str:
-        raise HAUnimplemented("This operation is not implemented.")
+    def __init__(self):
+        '''Init Method'''
+        super(ClusterNodeAddExecutor, self).__init__()
+        self._pcs_cluster_controller = PcsClusterController()
+        self._op = Output()
+        self._args = None
+
+    def parse_cluster_args(self) -> bool:
+        '''
+           Parses the command line args.
+           Return: argparse
+        '''
+        parser = argparse.ArgumentParser(prog='cluster add node')
+        parser.add_argument("cluster", help="Module")
+        parser.add_argument("add", help="action to be performed")
+        parser.add_argument("node", help="component on which action to be performed")
+        group = parser.add_mutually_exclusive_group(required='True')
+        group.add_argument('--nodeid', action='store', \
+                            help='ID of a node which needs to be added in a cluster')
+        group.add_argument('--descfile', action='store', \
+                            help='A file which describes the node to be added in a cluster', \
+                            type=argparse.FileType('r'))
+        parser.add_argument('--json', help='Required output fomat', action='store_true')
+        self._args = parser.parse_args()
+        return True
+
+    def validate(self) -> bool:
+        '''
+           Validates permission and command line arguments.
+           Exception: HAInvalidPermission
+           Return: argparse
+        '''
+        # Every CLI command will be an internal command now. So,
+        # we do not need this change. If required, can be revisited later.
+        #if not self.is_ha_user():
+        #    raise HAInvalidPermission('Not enough permissions to invoke this command')
+        if self.parse_cluster_args():
+            return True
+        return False
+
+    def execute(self) -> None:
+        '''
+           Execute CLI request by passing it to ClusterManager and
+           also displays an output
+        '''
+        # args = self.validate()
+        # TODO: Proper password to be sent
+        # TODO: Validate node_id and node description file by some means
+        if self._args.descfile:
+            self.parse_node_desc_file(self._args.descfile)
+        add_node_result_message = self._pcs_cluster_controller.add_node(self._args.nodeid, \
+                                    const.USER_HA_INTERNAL, 'abc')
```

Comment:
The userid and password seems to be hardcoded, why?

Reply:
Now this will be through command line args.

Reply:
yes